java-checks/src/main/java/org/sonar/java/checks/ShiftOnIntOrLongCheck.java

```diff
@@ -0,0 +1,130 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.check.BelongsToProfile;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.expression.ArrayAccessExpressionTreeImpl;
+import org.sonar.java.model.expression.AssignmentExpressionTreeImpl;
+import org.sonar.java.model.expression.IdentifierTreeImpl;
+import org.sonar.java.model.expression.LiteralTreeImpl;
+import org.sonar.java.model.expression.MethodInvocationTreeImpl;
+import org.sonar.java.model.expression.ParenthesizedTreeImpl;
+import org.sonar.plugins.java.api.tree.ArrayAccessExpressionTree;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
+
+import java.util.List;
+
+@Rule(
+  key = "S2183",
+  priority = Priority.CRITICAL,
+  tags = {"bug"})
+@BelongsToProfile(title = "Sonar way", priority = Priority.CRITICAL)
+public class ShiftOnIntOrLongCheck extends SubscriptionBaseVisitor {
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Kind.LEFT_SHIFT, Kind.LEFT_SHIFT_ASSIGNMENT, Kind.RIGHT_SHIFT, Kind.RIGHT_SHIFT_ASSIGNMENT);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    ExpressionTree target = null;
+    ExpressionTree shift = null;
+
+    if (tree.is(Kind.LEFT_SHIFT, Kind.RIGHT_SHIFT)) {
+      BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) tree;
+      target = binaryExpressionTree.leftOperand();
+      shift = binaryExpressionTree.rightOperand();
+    } else {
+      AssignmentExpressionTreeImpl assignmentExpressionTree = (AssignmentExpressionTreeImpl) tree;
+      target = assignmentExpressionTree.variable();
+      shift = assignmentExpressionTree.expression();
+    }
+
+    String identifier = getIdentifierName(target);
+    boolean expectLong = treeTypeIsLong(target);
+    boolean expectInt = treeTypeIsInt(target);
+    boolean shiftIsMinus = shift.is(Kind.UNARY_MINUS);
+
+    if (shift.is(Kind.UNARY_MINUS, Kind.UNARY_PLUS)) {
+      shift = ((UnaryExpressionTree) shift).expression();
+    }
+
+    if (isShiftEvaluable(shift, expectInt || expectLong)) {
+      String value = ((LiteralTreeImpl) shift).value();
+      long numberBits = (shiftIsMinus ? -1 : 1) * (expectInt ? Integer.parseInt(value) : Long.parseLong(value));
```

Comment:
Since your are anyway assigning it to a long, you don't care about parsing value with Integer.parseInt and can use directly Long.parseLong.